WordPress/Classes/ViewRelated/NUX/LoginEpilogueTableViewController.swift

```diff
@@ -108,40 +94,64 @@ extension LoginEpilogueTableViewController {
         if section == Sections.userInfoSection {
-            return 1
+            return 2
         }
 
         let correctedSection = section - 1
         let siteRows = blogDataSource.tableView(tableView, numberOfRowsInSection: correctedSection)
 
-        // Add one for the Connect Site row if shown.
-        return showConnectSite ? siteRows + 1 : siteRows
+        // Add one for Create new site cell
+
+        guard let parent = parent as? LoginEpilogueViewController else {
+            return siteRows
+        }
+
+        if siteRows <= 3 {
+            parent.hideButtonPanel()
+            return showCreateNewSite ? siteRows + 1 : siteRows
+        } else {
+            if !showCreateNewSite {
+                parent.hideButtonPanel()
+            }
+            return siteRows
+        }
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
 
         // User Info Row
         if indexPath.section == Sections.userInfoSection {
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: Settings.userCellReuseIdentifier) as? EpilogueUserInfoCell else {
-                return UITableViewCell()
-            }
-            if let info = epilogueUserInfo {
-                cell.stopSpinner()
-                cell.configure(userInfo: info)
-            } else {
-                cell.startSpinner()
-            }
+            if indexPath.row == 0 {
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: Settings.userCellReuseIdentifier) as? EpilogueUserInfoCell else {
+                    return UITableViewCell()
+                }
+                removeSeparatorFor(cell)
+                if let info = epilogueUserInfo {
+                    cell.stopSpinner()
+                    cell.configure(userInfo: info)
+                } else {
+                    cell.startSpinner()
+                }
 
-            return cell
+                return cell
+            } else if indexPath.row == 1 {
+                guard let cell = tableView.dequeueReusableCell(withIdentifier: Settings.chooseSiteReuseIdentifier, for: indexPath) as? LoginEpilogueChooseSiteTableViewCell else {
+                    return UITableViewCell()
+                }
+                removeSeparatorFor(cell)
+                return cell
+            }
         }
 
-        // Connect Site Row
-        if indexPath.row == lastRowInSection(indexPath.section) && showConnectSite {
-            guard let cell = tableView.dequeueReusableCell(withIdentifier: LoginEpilogueConnectSiteCell.defaultReuseID) as? LoginEpilogueConnectSiteCell else {
+        // Create new site row
+        let siteRows = tableView.numberOfRows(inSection: indexPath.section)
+        let threshold = 4 // 3 site rows + 1 create site row
+        if siteRows <= threshold && indexPath.row == lastRowInSection(indexPath.section) && showCreateNewSite {
```

Comment:
Hmm, this is doing a lot. I wonder if we can move it to a var with a more meaningful name?

Reply:
Fixed! 64fa99a